File: app/src/main/java/Model/Offer.java
package Model;

import com.google.firebase.Timestamp;
import com.google.firebase.firestore.Blob;

public class Offer implements Comparable<Offer> {

    private String categoryName, productId, offerId;
    private Timestamp startTime, endTime;
    private Blob offerPic;
    private float discountPercentage;
    public int type;

    public Offer() {
    }

    public String getOfferId() {
        return offerId;
    }

    public void setOfferId(String offerId) {
        this.offerId = offerId;
    }

    public Timestamp getStartTime() {
        return startTime;
    }

    public void setStartTime(Timestamp startTime) {
        this.startTime = startTime;
    }

    public Timestamp getEndTime() {
        return endTime;
    }

    public void setEndTime(Timestamp endTime) {
        this.endTime = endTime;
    }

    public Blob getOfferPic() {
        return offerPic;
    }

    public void setOfferPic(Blob offerPic) {
        this.offerPic = offerPic;
    }

    public float getDiscountPercentage() {
        return discountPercentage;
    }

    public void setDiscountPercentage(float discountPercentage) {
        this.discountPercentage = discountPercentage;
    }

    public String getProductId() {
        return productId;
    }

    public void setProductId(String productId) {
        this.productId = productId;
    }

    public String getCategoryName() {
        return categoryName;
    }

    public void setCategoryName(String categoryName) {
        this.categoryName = categoryName;
    }


    @Override
    public int compareTo(Offer o) {
        if (type == 1) {
            return (int) (o.getDiscountPercentage() - this.getDiscountPercentage());
        } else {
            return o.getStartTime().compareTo(this.getStartTime());
        }
    }
}

File: app/src/main/java/Model/Offer.kt
package Model

import com.google.firebase.Timestamp
import com.google.firebase.firestore.Blob

class Offer : Comparable<Offer?> {
    private var categoryName: String? = null
    private var productId: String? = null
    private var offerId: String? = null
    private var startTime: Timestamp? = null
    private var endTime: Timestamp? = null
    private var offerPic: Blob? = null
    private var discountPercentage = 0f
    var type = 0
    fun getOfferId(): String? {
        return offerId
    }

    fun setOfferId(offerId: String?) {
        this.offerId = offerId
    }

    fun getStartTime(): Timestamp? {
        return startTime
    }

    fun setStartTime(startTime: Timestamp?) {
        this.startTime = startTime
    }

    fun getEndTime(): Timestamp? {
        return endTime
    }

    fun setEndTime(endTime: Timestamp?) {
        this.endTime = endTime
    }

    fun getOfferPic(): Blob? {
        return offerPic
    }

    fun setOfferPic(offerPic: Blob?) {
        this.offerPic = offerPic
    }

    fun getDiscountPercentage(): Float {
        return discountPercentage
    }

    fun setDiscountPercentage(discountPercentage: Float) {
        this.discountPercentage = discountPercentage
    }

    fun getProductId(): String? {
        return productId
    }

    fun setProductId(productId: String?) {
        this.productId = productId
    }

    fun getCategoryName(): String? {
        return categoryName
    }

    fun setCategoryName(categoryName: String?) {
        this.categoryName = categoryName
    }

    override fun compareTo(o: Offer?): Int {
        return if (type == 1) {
            (o.getDiscountPercentage() - getDiscountPercentage()) as Int
        } else {
            o.getStartTime().compareTo(getStartTime())
        }
    }
}